fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1091,21 +1126,37 @@ public Builder setIncreaseLimitAfter(int increaseLimitAfter) {
              */
             @Nonnull
             public Builder setTimeLimitMilliseconds(long timeLimitMilliseconds) {
-                if (timeLimitMilliseconds < UNLIMITED_TIME) {
+                if (timeLimitMilliseconds < 0) {
                     timeLimitMilliseconds = UNLIMITED_TIME;
                 }
                 this.timeLimitMilliseconds = timeLimitMilliseconds;
                 return this;
             }
 
+            /**
+             * Set the time quota for a single transaction. If this quota is exceeds, the indexer will commit the
+             * transaction and start a new one. This can be useful to avoid timeouts while scanning a high number of records
+             * in each transaction.
+             * A non-positive value (default) implies unlimited quota.
+             * Note that this quota, if reached, will be exceeded by an Order(1) overhead time. Keeping some margins might be
+             * a good idea.
+             * @param timeQuotaMilliseconds the time limit in milliseconds
+             * @return this builder
+             */
+            @Nonnull
+            public Builder setTransactionTimeQuotaMilliseconds(long timeQuotaMilliseconds) {
+                this.transactionTimeQuotaMilliseconds = timeQuotaMilliseconds;
+                return this;
+            }
+
             /**
              * Build a {@link Config}.
              * @return a new Config object needed by {@link OnlineIndexer}
              */
             @Nonnull
             public Config build() {
-                return new Config(maxLimit, maxRetries, recordsPerSecond, progressLogIntervalMillis, increaseLimitAfter,
-                        maxWriteLimitBytes, timeLimitMilliseconds);
+                return new Config(maxLimit, initialLimit, maxRetries, recordsPerSecond, progressLogIntervalMillis, increaseLimitAfter,
```

Comment:
If `initialLimit > maxLimit`, should this do something like passing the `Math.min(initialLimit, maxLimit)` to the underlying `Config`? Or should the indexer set its initial limit to `Math.min(initialLimit, maxLimit)`?

Reply:
🤷 
There is a value in passing it 'as is' if the config loader inherits `initialLimit`, but only changes `maxLimit`.

Reply:
✅